src/main/java/apoc/number/Number.java

```diff
@@ -22,76 +22,97 @@
-  public Stream<StringResult> formatDouble(final @Name("number") double number, @Name("pattern") String pattern, @Name("language") String language) {
-    DecimalFormat format = buildFormatter(pattern, language);
-    return Stream.of(new StringResult(format.format(number)));
+  @Procedure("apoc.number.format.pattern")
+  @Description("apoc.number.format.pattern(number, pattern) yield value | format a long or double using a pattern and the default system language to produce a string")
+  public Stream<StringResult> formatByPattern(final @Name("number") Object number, final @Name("pattern") String pattern) {
+    return formatByPatternAndLanguage(number, pattern, null);
   }
   
-  @Procedure
-  @Description("apoc.number.parseAsLong(text, pattern, locale) yield value | parse a text using a (optional) pattern and for a (optional) locale to produce a long")
-  public Stream<LongResult> parseAsLong(final @Name("text") String text, @Name("pattern") String pattern, @Name("language") String language) throws ParseException {
-    DecimalFormat format = buildFormatter(pattern, language);
-    return Stream.of(new LongResult(format.parse(text).longValue()));
+  @Procedure("apoc.number.format.lang")
+  @Description("apoc.number.format.lang(number, lang) yield value | format a long or double using the default system pattern pattern and a language to produce a string")
+  public Stream<StringResult> formatByLanguage(final @Name("number") Object number, final @Name("lang") String lang) {
+    return formatByPatternAndLanguage(number, null, lang);
   }
-
-  @Procedure
-  @Description("apoc.number.parseAsDouble(text, pattern, locale) yield value | parse a text using a (optional) pattern and for a (optional) locale to produce a double")
-  public Stream<DoubleResult> parseAsDouble(final @Name("text") String text, @Name("pattern") String pattern, @Name("language") String language) throws ParseException {
-    DecimalFormat format = buildFormatter(pattern, language);
-     return Stream.of(new DoubleResult(format.parse(text).doubleValue()));
+  
+  @Procedure("apoc.number.format.pattern.lang")
+  @Description("apoc.number.format.pattern.lang(number, pattern, lang) yield value | format a long or double using a pattern and a language to produce a string")
+  public Stream<StringResult> formatByPatternAndLanguage(final @Name("number") Object number, final @Name("pattern") String pattern, final @Name("lang") String lang) {
+    validateNumberParam(number);
+    DecimalFormat format = buildFormatter(pattern, lang);
+    return Stream.of(new StringResult(format.format(number)));
+  }
+  
+  // Parse As Long
+  
+  @Procedure("apoc.number.parseAsLong")
+  @Description("apoc.number.parseAsLong(text) yield value | parse a text using the default system pattern and language to produce a long")
+  public Stream<LongResult> parseAsLong(final @Name("text") String text) throws ParseException {
```

Comment:
No keep them

Von meinem iPhone gesendet

> Am 26.08.2016 um 02:46 schrieb Lorenzo Speranzoni <notifications@github.com>:
> 
> In src/main/java/apoc/number/Number.java:
> 
> > -    DecimalFormat format = buildFormatter(pattern, language);
> > -     return Stream.of(new DoubleResult(format.parse(text).doubleValue()));
> > +  
> > +  @Procedure("apoc.number.format.pattern.lang")
> > +  @Description("apoc.number.format.pattern.lang(number, pattern, lang) yield value | format a long or double using a pattern and a language to produce a string")
> > +  public Stream<StringResult> formatByPatternAndLanguage(final @Name("number") Object number, final @Name("pattern") String pattern, final @Name("lang") String lang) {
> > +    validateNumberParam(number);
> > +    DecimalFormat format = buildFormatter(pattern, lang);
> > +    return Stream.of(new StringResult(format.format(number)));
> > +  }
> > +  
> > +  // Parse As Long
> > +  
> > +  @Procedure("apoc.number.parseAsLong")
> > +  @Description("apoc.number.parseAsLong(text) yield value | parse a text using the default system pattern and language to produce a long")
> > +  public Stream<LongResult> parseAsLong(final @Name("text") String text) throws ParseException {
> @jexp should I return a ResultObject so that we can merge parseAsLong and parseAsDouble into a more generic parse?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> -- 
> You received this message because you are subscribed to the Google Groups "neo4j-oss-build" group.
> To unsubscribe from this group and stop receiving emails from it, send an email to neo4j-oss-build+unsubscribe@neotechnology.com.
> To post to this group, send email to neo4j-oss-build@neotechnology.com.
> To view this discussion on the web visit https://groups.google.com/a/neotechnology.com/d/msgid/neo4j-oss-build/neo4j-contrib/neo4j-apoc-procedures/pull/123/r76349936%40github.com.